utils/src/main/java/io/neow3j/contract/NeoURI.java

```diff
@@ -0,0 +1,184 @@
+package io.neow3j.contract;
+
+import io.neow3j.utils.AddressUtils;
+import io.neow3j.utils.Strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NeoURI {
```

Comment:
For the parsing, I would create:

`public static NeoURI fromURI(String)`

or something like this.

You should search our code base for more examples.